src/app/models/user.server.model.ts

```diff
@@ -48,9 +48,31 @@ const compileUserModel = (db: Mongoose) => {
       type: Date,
       default: Date.now,
     },
+    contact: {
+      type: String,
+      validate: {
+        // Check if phone number is valid.
+        validator: function (value: string) {
+          const phoneNumber = parsePhoneNumberFromString(value)
+          if (!phoneNumber) return false
+          return phoneNumber.isValid()
+        },
+        message: (props) => `${props.value} is not a valid mobile number`,
+      },
+    },
     betaFlags: {},
   })
 
+  // Hooks
+  // Unique key violation custom error middleware
+  UserSchema.post<IUserSchema>('save', function (err, doc, next) {
+    if (err.name === 'MongoError' && err.code === 11000) {
```

Comment:
can link to the documentation for where to find this magic number?

Reply:
Thanks. I'll probably use `https://www.npmjs.com/package/mongoose-unique-validator` instead!